geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/metadata/package.scala

```diff
@@ -55,15 +58,20 @@ package object metadata {
   case class PartitionConfig(
       name: String,
       action: PartitionAction,
-      files: Set[StorageFile],
+      files: Seq[StorageFile],
       count: Long,
-      envelope: EnvelopeConfig,
+      envelope: Seq[Double],
       timestamp: Long) {
 
     def +(other: PartitionConfig): PartitionConfig = {
       require(action == PartitionAction.Add, "Can't aggregate into non-add actions")
       val ts = math.max(timestamp, other.timestamp)
-      PartitionConfig(name, action, files ++ other.files, count + other.count, envelope.merge(other.envelope), ts)
+      val env = if (envelope.isEmpty) { other.envelope } else if (other.envelope.isEmpty) { envelope } else {
+        val Seq(xmin1, ymin1, xmax1, ymax1) = envelope
+        val Seq(xmin2, ymin2, xmax2, ymax2) = other.envelope
+        Seq(math.min(xmin1, xmin2), math.min(ymin1, ymin2), math.max(xmax1, xmax2), math.max(ymax1, ymax2))
+      }
```

Comment:
This logic makes sense.  Should it be here? 

Definitely non-blocking, I'm just surprised that it hasn't been needed before here or elsewhere.

Reply:
i don't think we store envelopes as untyped Seqs like this anywhere else... the impetus here is to make the json rendering as compact as possible.